static/app/views/dashboards/widgetBuilder/buildSteps/thresholdsStep/thresholdsStep.tsx

```diff
@@ -0,0 +1,157 @@
+import styled from '@emotion/styled';
+
+import CircleIndicator from 'sentry/components/circleIndicator';
+import FieldWrapper from 'sentry/components/forms/fieldGroup/fieldWrapper';
+import NumberField, {NumberFieldProps} from 'sentry/components/forms/fields/numberField';
+import {t, tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import theme from 'sentry/utils/theme';
+
+import {BuildStep} from '../buildStep';
+
+type ThresholdsStepProps = {
+  onChange: (maxKey: ThresholdMaxKeys, value: string) => void;
+  thresholdsConfig: ThresholdsConfig | null;
+};
+
+type ThresholdRowProp = {
+  color: string;
+  maxInputProps: NumberFieldProps;
+  minInputProps: NumberFieldProps;
+  maxKey?: ThresholdMaxKeys;
+  onChange?: (maxKey: ThresholdMaxKeys, value: string) => void;
+};
+
+export enum ThresholdMaxKeys {
+  MAX_1 = 'max_1',
+  MAX_2 = 'max_2',
+}
+
+type ThresholdMaxValues = {
+  [K in ThresholdMaxKeys]?: number;
+};
+
+export type ThresholdsConfig = {
+  max_values: ThresholdMaxValues;
+  unit: string | null;
+};
+
+const WIDGET_INDICATOR_SIZE = 15;
+
+function ThresholdRow({
+  color,
+  minInputProps,
+  maxInputProps,
+  onChange,
+  maxKey,
+}: ThresholdRowProp) {
+  const handleChange = (val: string) => {
+    if (onChange && maxKey) {
+      onChange(maxKey, val);
+    }
+  };
+
+  return (
+    <ThresholdRowWrapper>
+      <CircleIndicator color={color} size={WIDGET_INDICATOR_SIZE} />
+      <NumberField {...minInputProps} inline={false} />
+      {t('to')}
+      <NumberField onChange={handleChange} {...maxInputProps} inline={false} />
```

Comment:
This should also have some min value configuration as well, right? Because currently I can set the threshold to lower than what my min was and I think the minimum should be the threshold + 1 or something. I noticed I can set negative values so I think if it's an easy change we should restrict that

Reply:
The min prop on the NumberField prop doesn't seem to work. I'll add the negative validation to the backend where I'll also check for sorting.